src/mixins/amalgamation-mixin.ts

```diff
@@ -429,7 +429,7 @@ export default class AmalgamationMixin extends Vue {
     }
 
     // set new resulting business name and legal type
-    // and update resources (since legal  type may have changed)
+    // and update resources (since legal type may have changed)
     this.setNameRequestApprovedName(business.name)
     this.setEntityType(business.legalType)
     this.updateResources()
```

Comment:
This is `updatePrepopulatedData()` and this code overrides the previous legal name and legal type. This is called when:
- a draft amalgamation is restored (see filing-template-mixin.ts below)
- the user marks a different business as holding/primary
- the user is submitting this filing (see Actions.vue above)